### PR TITLE
Support building packages on non-x86 platform

### PR DESCRIPTION
**Pull request by syuu1228**
Allow users to build unofficial packages for non-x86 platform.

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -1260,7 +1260,8 @@ def configure_seastar(build_dir, mode):
 
     dpdk = args.dpdk
     if dpdk is None:
-        dpdk = mode == 'release'
+        if platform.machine() == 'x86_64':
+            dpdk = mode == 'release'
```
Comment (avikivity) [inline configure.py:1264]: This leaves `dpdk` as None sometimes. It's better to keep it true/false (maybe dpdk = platform.machine() == 'x86_64' and mode == 'release').

File: dist/redhat/scylla.spec [hunk 2]
```diff
@@ -21,6 +21,10 @@ Obsoletes:	scylla-server < 1.1
 %undefine _unique_build_ids
 %global _no_recompute_build_ids 1
 
+%ifnarch x86_64
+%global _missing_build_ids_terminate_build 0
+%endif
```
Comment (avikivity) [inline dist/redhat/scylla.spec:26]: We'll have to fix it. Please file an issue so we don't forget and copy @espindola. Meanwhile we can live without the build-id.

File: dist/debian/control.template [hunk 3]
```diff
@@ -16,7 +16,7 @@ Replaces: %{product}-server (<< 1.1)
 Conflicts: %{product}-server (<< 1.1)
 
 Package: %{product}-server
-Architecture: amd64
+Architecture: %{arch}
```
Comment (avikivity) [inline dist/debian/control.template:19]: I think you can write `any` here: https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-architecture

**Comment by espindola** [comment]
Avi Kivity <notifications@github.com> writes:

> +%ifnarch x86_64
> +%global _missing_build_ids_terminate_build 0
> +%endif
>
> We'll have to fix it. Please file an issue so we don't forget and copy @espindola. Meanwhile we can live without the build-id.

Could we at least have a comment saying what is mising a buildid on non
x86_64? Do we know why?

Cheers,
Rafael

**Comment by avikivity** [comment]
> Do we know why?

That's why we copied you.

File: dist/redhat/scylla.spec (hunk 2 above)
Comment (syuu1228) [reply dist/redhat/scylla.spec:26]: made #7046, also find out patchelf breaking build-id

**Comment by syuu1228** [comment]
@avikivity dropped build-id patch created #7046 instead, fixed dpdk patch and debian patch. Please review again

File: configure.py [hunk 4]
```diff
@@ -1262,7 +1262,7 @@ def configure_seastar(build_dir, mode):
 
     dpdk = args.dpdk
     if dpdk is None:
-        dpdk = mode == 'release'
+        dpdk = platform.machine() == 'x86_64' mode == 'release'
```
Comment (avikivity) [inline configure.py:1265]: Missing `and`

File: dist/debian/python3/control.template [hunk 5]
```diff
@@ -8,7 +8,7 @@ Standards-Version: 3.9.5
 Rules-Requires-Root: no
 
 Package: %{product}-python3
-Architecture: amd64
+Architecture: any
```
Comment (avikivity) [inline dist/debian/python3/control.template:11]: Don't forget to patch scylla-python3.git too (and let's revive that patch to use it)

File: configure.py (hunk 4 above)
Comment (syuu1228) [reply configure.py]: @avikivity Just fixed.

File: dist/debian/python3/control.template (hunk 5 above)
Comment (syuu1228) [reply dist/debian/python3/control.template:11]: okay

**Pull request merged**